src/data/image.ts

```diff
@@ -35,29 +66,70 @@ const streamToS3 = (
   return pass;
 };
 
-export async function resizeImage(
-  imageUrl: URL,
-  size: number
+export async function uploadUserImage(
+  userSlug: string,
+  imageUrl: URL
 ): Promise<string> {
-  const s3 = new S3({
-    accessKeyId: process.env.ACCESS_KEY_ID,
-    secretAccessKey: process.env.SECRET_ACCESS_KEY,
-    region: process.env.REGION,
-  });
+  const bucketPrefix = "team-full/";
+  const Bucket = process.env.IMAGE_BUCKET as string;
 
   if (!imageUrl.href.startsWith("https://drive.google.com")) {
     throw new Error(`Error processing ${imageUrl.href}, not a drive url`);
   }
 
   const imageId = imageUrl.href.split("/file/d/")[1]?.split("/")[0];
-
   if (!imageId) {
     throw new Error(
       `Invalid formatted google drive image url: ${imageUrl.href}`
     );
   }
 
-  const resizer = sharp().rotate().resize(size, size);
+  /**
+   * Used to determine the uniqueness of images. Any images w/ the same
+   * hash for the same user won't be overwrote. If changes to the image
+   * processing are done you can update the salt value to regenerate all
+   * the hashes.
+   */
+  const imageUrlHash = hash(imageUrl.toString() + "salt2");
+  const userImageName = `${bucketPrefix}${userSlug}-${imageUrlHash}`;
+
+  const [ErrorAccessingS3Images, S3Images] = await to(
+    listS3Images(bucketPrefix)
+  );
```

Comment:
[await-to-js](https://www.npmjs.com/package/await-to-js) is probably hands down my most used tool for writing async logic.